setupdoxyfile.py

```diff
@@ -0,0 +1,34 @@
+import yaml
+import os
+
+
+def main():
+  f = open("config.yml", "r")
+  config = yaml.load(f, Loader=yaml.Loader)
+  os.system(
```

Comment:
Not strictly necessary, but it's better overall to use templates in Python internally rather than depending on `os.system()`. Makes the code platform-agnostic.

This part could be replaced by Python string template (have a config/Dockerfile.tmpl, for instance, with template strings for PROJECT_NAME, OUTPUT_DIRECTORY, etc.), and then using the python `subprocess` module to invoke `doxygen`.

Not urgent, and can be a separate follow-up commit.

Reply:
Also, just curious, why have a separate branch for the docs? Would probably make things simpler to have docs be a directory in the main branch itself. With two branches, if you need to make some changes which span both branches, you need separate pull requests, and if the code refers to the branch names, testing becomes a bit more complicated too. For instance, if you have a change that refers to the `docs` branch from `main`, and you have a test branch called `docs-xyz`, you will need to refer to `docs-xyz` in your tests, and change that back to `docs` before the final merge.

Reply:
We use Github Pages for those docs, I used docs branch thinking that this the best approach by [Github Pages ](https://docs.github.com/en/pages/getting-started-with-github-pages/configuring-a-publishing-source-for-your-github-pages-site). 

Looking at your comment and reading more about Github Pages configs, I think you have a valid point. The whole of point of this CI/CD is to eliminate steps not to add more.

Will have to think more about this approach and maybe use /docs directory on main, as Franklin did with C# and Android SDKs.